There are many difficulties facing the researcher who wishes to have a
high-level view of the ``state-space'' of the human transcriptome, or the
complete set of stable states allowable by the transcriptional regulatory
network of the human genome \cite{XXX}.

The first of these difficulties is the assembly of measurements of gene
transcription across as many phenotypic conditions as possible.  The second
difficulty is pre-processing those measurements so that they may be used
together in a single meta-analysis.  A solution to the problem of
pre-processing was presented by XXX, et al \cite{genelogic}, and an
implementation of that method was incorporated into Celsius, a microarray
database warehouse system previously published by these authors \cite{celsius}.
The scope of Celsius was to identify, amalgamate, and preprocess all publicly
available Affymetrix microarray data.  While such a data set is interesting for
investigations of low-level studies of microarray behavior, it is not by itself
particularly useful for understanding the structure of transcriptome state
space.  The problem is not that the state space cannot be observed, but rather
that it is lacking labels.  What is needed is a mapping between anotomical,
physiological, and cellular structures phenotypes and regions of the space.

We present here our construction of an automated system that is able to assign
labels to some of the space described.  We demonstrate the usage of flat
(random forest) and hierarchical (pachinko machine) classification algorithms
to assign high-level phenotypic labels to many samples in the Celsius data
warehouse.


%#####
PMID_17453049	Revealing static and dynamic modular architecture of the eukaryotic protein interaction network.
PMID_18007649	The road to modularity.
PMID_17353930	Network-based prediction of protein function.
