%$Id: gmod_web_03-25-07.tex,v 1.1 2007/04/04 22:06:26 boconnor Exp $

% Abstract

\section{Abstract}

% No more than 250 words, with second para for web address.  \cite{gbrowse}

The GMODWeb project is a software framework designed to speed the development
of websites for Model Organism Databases (MODs).  GMODWeb is part of the larger
Generic Model Organisms Database (GMOD) initiative which provides
species-agnostic software tools and data models for representing curated model
organism system data.  Users of GMODWeb can browse and search through many
different data types including genomic features and annotations, stocks,
literature references, and genomic maps. It is also integrated with other GMOD
tools such as GBrowse, AmiGO, and Textpresso. To assist development of new MOD
sites end-to-end examples of fully customized and integrated GMODWeb instances
for the \emph{H.  sapiens} and \emph{S.  cerevisiae} genomes have been created.
The recently inaugurated ParameciumDB also uses GMODWeb as its core website
technology for presenting key data of interest to this MOD community.  GMODWeb
is built on the flexible Turnkey web framework and is freely available under an
open source license from \url{http://turnkey.sourceforge.net}.  User
documentation, support forums, and source downloads are available at this site
while pre-packaged versions for Linux distributions are downloadable at the
Biopackages repository for bioinformatics software
(\url{http://biopackages.net}).


% Introduction

\section{Introduction}

Model Organism Databases (MODs) are built around the information needs of
scientists working on a single model organism or group of closely related
organisms.  Examples of MODs include Flybase (\url{http://www.flybase.org})
\cite{PMID_16381917}, Wormbase (\url{http://www.wormbase.org}) \cite{wormbase}, the
Mouse Genome Informatics (MGI) Database (\url{http://www.informatics.jax.org})
\cite{mgid}, the \emph{Saccharomyces} Genome Database (SGD)
(\url{http://www.yeastgenome.org}) \cite{sgd}, Gramene
(\url{http://www.gramene.org}),  a monocot genomics database
\cite{PMID_16381966}, and ParameciumDB (\url{http://paramecium.cgm.cnrs-gif.fr})
\cite{ParameciumDB}.  MODs provide scientists with access to information about
genomic structure, phenotypes, and mutations along with large-scale datasets
such as those generated by gene microarray experiments, SNP analyses, or
protein-protein interaction studies.  A key concern for any MOD is to provide
well-designed and convenient community tools for accessing this information.
All MODs create websites to fulfill these needs, an expensive and
time-consuming prospect. As many more model organisms are sequenced the costs,
in terms of both time and funds, of independently developing schemata and
web-based tools will become prohibitive.

Recognizing this duplication of work, the NIH and the USDA Agricultural
Research Service funded the Generic Model Organisms Database (GMOD) project
with the goal of developing flexible applications that can be used across all MODs.
The result is a collection of database and web tools that can be mixed
and matched to meet the requirements of new MODs.
%These tools address needs for representing and visualizing both genetic and
%physical maps, storing and searching literature curations, and browsing
%ontological annotations, along with an extensible architecture for a MOD
%website development.
To date, this effort has produced several high-profile components. A generic
and modular relational database schema, called Chado, provides the core
mechanism to store genomic features, information on gene function, genomic
diversity data, literature references, and other common data types.  Other
popular GMOD tools include Apollo \cite{apollo}, an application for genomic
curation, GBrowse \cite{gbrowse}, a web-based genomic browser that can
effectively display genomic features across megabases of sequence, and
Textpresso \cite{textpresso}, a web tool for literature archiving and
searching. While several solutions exist for representing genome annotation
data on the web, such as Ensembl \cite{ensembl} and the UCSC genome browser
\cite{PMID_12045153}, no solution exists for representing the full variety of
data types needed for a MOD.  In this paper we describe GMODWeb, a flexible and
extensible framework for creating a MOD website that integrates with other GMOD
tools and accommodates all the data types needed for a model organism database.

% NOTE: Lincoln removed this from Draft3
%Every MOD needs a web presence to disseminate information about the model
%organism.  Other web frameworks for presenting biological data exist, such as
%GenBank \cite{genbank} and Ensembl \cite{ensembl}, but their focus differs
%from what is needed for a MOD.  While GenBank focuses on genomic feature
%archiving and Ensemble on whole genome annotations, a MOD website need to
%present annotations and data for just one organism and present it in a way
%that is helpful for the MOD community as a whole.  The emphasis for these
%species-specific sites is more centrally placed on genome annotation and the
%web interfaces to these MODs need to primarily share these annotations with
%end users. The GMODWeb application has been designed closely around the common
%GMOD Chado schema.  Since the schema forms the basis for all the MODs, a web
%tool that accurately represents the schema and allows an end user to browse
%the information contained within is key.  GMODWeb has been specifically
%created to allow end users to explore the features and annotations stored in a
%Chado database and to link out to other websites and tools including
%BlastGraphic \cite{blast} \cite{blastgraphic}, Textpresso \cite{textpresso},
%GBrowse \cite{gbrowse}, among others.  To date, two GMODWeb example sites have
%been created, for human and yeast.  These are available as part of a sample
%GMODWeb install and will help future MODs by providing concrete examples of a
%GMODWeb setup that can be directly adapted and reconfigured for other model
%organism databases.  Already, ParameciumDB \cite{parameciumdb}
%(\url{http://paramecium.cgm.cnrs-gif.fr) has used these examples to create a
%GMODWeb instance.


\section{Results}

\subsection{GMODWeb Architecture}

GMODWeb is based on the Turnkey (\url{http://turnkey.sourceforge.net}) site
generation and rendering framework which consists of two distinct components.
The first is a code creation tool (Turnkey::Generate) that produces a Model
View Controller (MVC) based website given a database schema file \cite{mvc}. In
the case of GMODWeb this is the Chado schema.  The second Turnkey component
(Turnkey::Render) is a page-rendering module that links the generated MVC code
to an Apache webserver (\url{http://apache.org}). This portion of the Turnkey
framework uses a collection of open source perl modules and the popular
mod\_perl webserver plugin (\url{http://perl.apache.org}). Each Turnkey
component is used in a different phase of website construction.  While the
MVC generator automates the creation of most site code, the page-rendering
module handles the response to user requests received by the webserver.

Turnkey, and GMODWeb by extension, is strictly divided into MVC layers.  This
style of abstraction is a useful tool for organizing a web application into
manageable layers and improves the overall organization of the software.
Likewise, the active code generation approach used by Turnkey, which is similar
to the Object Management Group's (\url{http://www.omg.org/mda}) Model Driven
Architecture (MDA) proposal, is especially useful for the GMODWeb project
because underlying changes in the data model are quickly and easily integrated
into the application \cite{mda}.  For example, the inclusion of new database
modules in Chado can be easily accommodated by regenerated the Turnkey site
from the database schema file. GMODWeb is produced by simply applying
customizations, a GMODWeb "skin", to the Turnkey website auto-generated using
the Chado schema. This decoupling of user interface customization from
underlying data structure makes the GMODWeb application easy to extend,
customize, and maintain. Figure 1 shows the close relationship between GMODWeb
and Turnkey.

%FIXME: there's a lot more that can be added here with more talk about MDA and
%MVC being top on the list.  

\subsection{GMODWeb Site Generation and Rendering}

The creation of a Turnkey site, such as GMODWeb, beings with a SQL schema file
used to define the tables in a database and how they relate to each other.
This file is abstracted into relationships between objects forming a Directed
Graph (DG).  Turnkey::Generate uses the perl module SQL::Translator to perform
this conversion from a SQL schema file to a DG object model
(\url{http://sqlfairy.sourceforge.net}).  For example, in the Chado schema a
{\texttt feature} table stores information about genomic features such as mRNAs or
genes.  This table is linked to many other tables, such as the {\texttt synonym}
table via the {\texttt feature\_synonym} table.  The Turnkey::Generate script
creates objects representing each table ({\texttt feature}, {\texttt synonym} and {\texttt
feature\_synonym}) and their individual data fields.  It then creates links
between these objects to mimic the relationships encoded by the schema, in this
case linking the {\texttt feature} and {\texttt synonym} tables.

Using the relationships encoded by the DG, Turnkey::Generate produces an MVC
mod\_perl website.  
%An MVC-based application is desirable because it segments an application into
%components that communicate with a datasource, others that render the user
%interface, and controller components that manage the flow of information.
%This style of programming is commonly used for web development and the
%division into these application layers provides abstraction and flexibility.
Each layer of the MVC framework is created using Template::Toolkit templates.
The model layer, which handles the flow of information to and from the
underlying database, is created using a template to produce Class::DBI-based
objects (\url{http://wiki.class-dbi.com}).  Class::DBI is a convenient tool to
connect and retrieve information from the database because it abstracts complex
SQL queries into easy-to-use object calls.  Controller objects, called atoms in
the Turnkey framework, wrap the model objects and provide an abstraction
between the view and model objects.  They also include the logic necessary to
bring these two layers together.  The view layer is, itself, implemented in
Template::Toolkit and uses HTML with embedded tags to extract information from
controller objects for display to the end user.  Turnkey::Generate also creates
the Turnkey.xml controller document that describes how model and view objects
are to be combined by the atom controller objects. Figure 2a illustrates the
MVC-based architecture created with the Turnkey::Generate software.

Once created, the output of Turnkey::Generate is configured to work in an
Apache server using the mod\_perl framework.  The process of rendering a page
is handled by Turnkey::Render. When a user requests a certain URL, the
Turnkey.xml document is examined by Turnkey::Render and the appropriate
Class::DBI model and controller atom objects are instantiated. For example, the
{\texttt feature} table described previously has an entry in this XML linking it to
the {\texttt synonym} table through the {\texttt feature\_synonym} table.  This
provides Turnkey::Render with enough information to create atom and model
objects for both the feature and synonym tables.  Following this, the
appropriate template view objects are created and Turnkey::Render uses the atom
controller objects to manage the handoff of objects and template files to the
Template::Toolkit engine for rendering.  The resulting HTML output is then
returned to the client (Figure 2b).

\subsection{GMODWeb Customization}

Customization is an important feature that all MODs require in their web
interfaces.  To accommodate this, key design features were integrated into the
Turnkey framework affecting both the site generation and page rendering
processes.  These include template customization through overriding and
Cascading Style Sheet-based (CSS) layouts (\url{http://www.w3.org/Style/CSS}).
Template overriding provides the ability for MOD developers to create a
customized look and feel for a given type of information being displayed in a
GMODWeb site. For example, the default feature page in GMODWeb was swapped out
for a custom templates that included a GBrowse panel if the feature had a
genomic location, as is the case for a gene or an mRNA transcript feature.
These custom templates would normally be overwritten in an MDA web framework
but Turnkey allows site designers to create and persist these modifications.
In addition to template customization, layout and styling in a Turnkey site is
accomplished with flexible CSS documents, allowing the MOD developer to
dramatically change the look and feel of the entire site. Not only can colors
and fonts be changed, but element layouts can be reordered.

A combination of these customizations can be grouped together into a "skin"
which can easily be parameterized and switched on the fly.  This makes it
possible for a MOD website to be context dependent and support a "print" view or
completely different color scheme with the same underlying website and
database. For example, a clade-oriented database that provides information on
12 different beetle species could apply a different page color to each species
to avoid user confusion.  GMODWeb was created by taking a Turnkey website
generated from the Chado database schema and applying these types of
customizations to the codebase (Figure 1).   

Demonstration GMODWeb sites have been created for \emph{H. sapiens} and
\emph{S. cerevisiae} and include the basic functionality associated with a
typical MOD's homepage. These sites illustrate the common layout for a Turnkey
website and show the effects of a customized GMODWeb skin. The sample websites
include the ability to search by features and controlled vocabulary (CV) terms
indexed from the underlying Chado database using the open source search engine
Lucene (\url{http://lucene.apache.org}).  Since many data types in Chado are
annotated and linked together through CV terms using various ontologies, such
as the Gene Ontology (GO) \cite{PMID_10802651}, it was important to be able to
query by both data types.  Search results will take an end user to either a
feature or CV term page rendered using customized GMODWeb templates. 

% NOTE: Lincoln removed
%The resulting Turnkey framework is designed to be completely auto-generated yet
%to provide a flexible skin approach to change the look and feel of a site.
%Customized view templates, along with customized CSS and images, can be
%combined to form a skin.  These skins reside in their own directory and are
%persistent when a site is subsequently regenerated.  This allows customized
%interfaces to persist while the underlying database changes.

Browsing a feature reveals several customizations to the default templates.
Figure 3 shows a typical gene feature page using the GMODWeb skin from the
ParameciumDB MOD website.  In this example, the basic layout of a Turnkey page
is evident: the item being rendered, in this case a row from the {\texttt feature}
table, is present as the major content panel while linked tables are
represented as minor panels on the left-hand side.  For this gene feature, two
types of linked data were presented on the left: external references (via the
{\texttt feature\_dbxref} table) and relationships to other features in the
database (via the {\texttt feature\_relationship} table).
%, and results of computational analysis (via the {\texttt featureloc} table).
Customizations of links and panel headings in both the major and minor panels
are shown in this example as well.

% fixme: could add more about link and header cust.

Further customization was used in the major panel to organize information about
the gene feature in an intuitive and helpful way.  Related content, such as GO
term annotations, genomic location, synonyms, and other information, was
included as a summary.  The Turnkey framework's flexibility allows custom
template authors to easily extract this information using the underlying
Class::DBI model objects. In this customized template, simple nested method
calls on the feature model object were used to extract linked information
such as synonyms.  Together these modifications have created a gene page that
can be leveraged across MODs and provide many of the key pieces of information
about genomic features that end users will require. Turnkey pages also contain
an edit link which provides a limited but useful facility for editing record
data.  Authentication is provided by standard HTTP access controls in Apache.

\subsection{GMODWeb Integration}

The example in Figure 3 shows how GMODWeb's templates were directly integrated
with other GMOD projects.  In this page, a GBrowse instance was embedded and
provided not only a graphical view of the genomic neighborhood but also linked
out to nearby genes and other annotations.  In addition to GBrowse, the sample
GMODWeb sites for \emph{H. sapiens} and \emph{S. cerevisiae} include
integration with Textpresso for literature tracking, BlastGraphic for
performing Blast analysis, and AmiGO for controlled vocabulary term
visualization. These dependencies, which are available from the Biopackages
software repository, have been pre-configured to work with the GMODWeb
demonstration sites. Packaging the sample applications and their dependencies
makes installation and configuration a quick and easy task for site developers
and jumpstarts the process of setting up new MOD websites.

In addition to web interfaces, GMODWeb also provides Simple Object Access
Protocol (SOAP) (\url{http://www.w3.org/TR/soap}) bindings for accessing data in an
automated, programmatic way.  This web services approach is designed to allow
savvy end users to interact directly with the underlying GMOD Chado database,
affording bulk access to features contained within the database.  Providing
this tool for GMODWeb's model objects makes data access platform agnostic so
developers can interact with the service using the language of their choice.
Apache2::SOAP (\url{http://search.cpan.org/~rkobes/Apache2-SOAP-0.72}) was used
to bind Class::DBI-based model objects to a SOAP interface.  Unlike XML genome
feature annotation services, such as the Distributed Annotation System (DAS)
\cite{PMID_11667947}, the SOAP bindings present low-level interfaces to
database tables. This SOAP interface is pre-configured and immediately
available for all MOD sites based on GMODWeb.

% FIXME: I need to combine this with the paragraph later on describing SOAP

%The Turnkey-based GMODWeb can be used for more than web-page rendering.
%Because of the flexible nature of the skins approach, and the general utility
%of the Apache environment, two developer-oriented services are available
%through all GMODWeb sites.  First, skins can be created to output XML rather
%than HTML, enabling developers and end users to retrieve information about
%genomic features programmatically.  Second, Apache::SOAP provides automated
%SOAP bindings to the underlying model layer \cite{apache_soap}.  This allows
%savvy developers to directly access a MOD through an easy-to-use and platform
%agnostic web services interface.  These abilities compliment other technologies
%already in use, such as XORT \cite{xort} for producing ChadoXML and DAS
%\cite{PMID_11667947} for producing feature-centric XML documents, and provide
%the means to extend a GMODWeb MOD site beyond simply a web interface. 

\subsection{Case Study: Creating a New MOD Website with GMODWeb and Turnkey}

Paramecium, a unicellular eukaryote that belongs to the ciliate phylum, has
served as a genetic model organism for over half a century and is also widely
used to teach biology. The genome of \emph{Paramecium tetraurelia} was recently
sequenced and annotated at the Genoscope French National Sequencing Center
\cite{ptgenome}. In anticipation of public release of the data from the
sequencing initiative, a project was started in 2005 to develop a Paramecium
community MOD, ParameciumDB. Its immediate objectives were to integrate
the genome sequence and annotations with available genetic data and 
coordinate the manual curation of the gene models by members of the research
community. Ultimately, ParameciumDB should provide a useful resource for the
classroom as well.

GMOD's Chado database schema was well suited for this project because of its
genetic module, which ensured the integration of both the genetic and sequence
data, and its support for describing phenotypes using controlled vocabulary
terms. Another important factor in choosing the GMOD toolkit for ParameciumDB
was the availability of Turnkey and GMODWeb to generate the MOD's website,
since it was anticipated that this would be the most difficult part of the project.

GMODWeb was first tested on a generic installation of Chado, populated with
published data from a previously sequenced and annotated Paramecium chromosome
\cite{Zagulski}.  The next step was modeling the genetic data, which involved
writing a stock sub-module for the Chado genetic module to make it possible to
incorporate data about Paramecium stock collections. Since the Chado database
schema was modified, Turnkey::Generate was used to create a custom website for
ParameciumDB.

The last and most time-consuming step for building ParameciumDB was
customization of the auto generated website layout. The overall design of the
site components (header, footer, feature page, etc) was achieved using
templates and CSS modifications of the GMODWeb skin, resulting in a custom
ParameciumDB skin.  Within the auto-generated view code, the
feature-relationship atom object was modified to make it possible to recover the
complete hierarchy of relationships (e.g.  gene $\rightarrow$ mRNA
$\rightarrow$ exon) from the top-level gene feature, even if the feature page
being rendered concerned a feature type lower in the hierarchy. Additional
static content was added to the site using Microsoft Frontpage including a help
section, project documentation, and announcements
(\url{http://office.microsoft.com/frontpage}).  Finally, commonly used applications
were integrated into the ParameciumDB by linking to other bioinformatics tools,
such as NCBI's BLAST tool \cite{blast}, and to forms for data submission by the
community.

The templates for pages within ParameciumDB were customized using many of the
ideas taken from the GMODWeb sample sites and, in particular, the layout of the
sample gene page.  The elegance of the Turnkey-based MVC site was most apparent
at this level: customization of ParameciumDB was focused on the template view
layer while relatively few changes were required to either the model or
controller objects.  The bulk of the code produced for ParameciumDB was
automatically generated and untouched by the customization process. This freed
developers to work on the effective presentation of MOD data rather than low
level database access or website rendering code.  


\section{Discussion}

Model organism databases (MODs) gather together biologic information on a
variety of important organisms for the scientific community. A key concern for
any MOD is to provide well-designed and easily accessible tools for sharing
this information.  The GMODWeb project was started to provide a simple and
generic solution for quickly creating new MOD websites using the Turnkey
framework.  GMODWeb, by running directly off the flexible and extensible Chado
schema, can accommodate the wide variety of data types and usage patterns that
model organism communities require.  It offers both a clean MVC framework and
pre-built sample websites configured to work with other GMOD tools. Together
these features can greatly streamline the process of new MOD website
development. 

One of the challenges for any framework based on Model Driven Architecture
techniques is balancing auto-generated code tied to a particular underlying
schema with customized layouts for creating compelling and effective user
interfaces.  Turnkey, the technology underling GMODWeb, attempted to solve this
limitation by providing a mechanism for bundling specialized skins with an
auto-generated website.  Since most of the website is automatically created,
designers can focus on the quality of the user interface and not on the
underlying rendering code.  In GMODWeb this translated to extensive
customization of the default feature templates. With the adoption of GMODWeb by
the ParameciumDB project, we anticipate incorporating UI changes and
customization into GMODWeb for the benefit of all future GMODWeb-based MODs.
As additional MODs adopt GMODWeb, we envision the availability of a large
library of site-specific customized templates, which can be adopted, altered
and expanded by subsequent MOD projects.

The rapidity with which ParameciumDB was built, by a very small development
team, is encouraging. For this MOD, data modeling was much more time-consuming
than building the GMODWeb-based website.  In fact, the only difficulty
encountered in implementation of ParameciumDB was not with GMODWeb or Turnkey
\emph{per se}, but with the installation and tuning of mod\_perl and the Apache
web server for use in a production environment. The current availability of
GMODWeb sample sites and installation dependences as pre-compiled software
packages on Biopackages should make this part of a new project much easier for
future MODs. 

% FIXME: this paragraph may be a little repetative with the first
As the model organism database community continues to expand, there will be an
increased need to leverage existing tools to store, query, and present MOD
data.  The GMOD project was created to engineer generic tools to meet these
needs.  GMODWeb was designed to quickly create MOD websites based on the easy
to use, customize, and update Turnkey framework. A GMODWeb MOD site provides
not only the ability to browse and search MOD data but it also forms a key link
to other tools.  As future applications and components become available,
GMODWeb will continue to act as a natural point of integration and a central
hub for the display of MOD data to end users.


\section{Methods}

\subsection{Availability}

Turnkey and GMODWeb are both available as a source code files from Turnkey's
website (\url{http://turnkey.sourceforge.net}) and as pre-compiled packages for
various Linux distributions from the Biopackages repository for bioinformatics
software (\url{http://biopackages.net}).  All dependencies are provided using
the Red Hat Package Manager (RPM) (\url{http://www.rpm.org}) and integration
with other programs, such as GBrowse, is accomplished using this same package
management system.  When a MOD installs GMODWeb via RPMs, a pre-configured
GBrowse, Blast server, Textpresso, and other applications are installed and
configured to work within GMODWeb immediately. Table 1 shows the software
dependencies for GMODWeb, all of which are available either from specific Linux
distributions or through Biopackages.

% FIXME: add table 1

\subsection{User Feedback}

The GMODWeb project is supported with help from the larger open source
development community.  Information on installation, troubleshooting, and
optimization can be found on the Turnkey website at
\url{http://turnkey.sourceforge.net}.  MODs setting up GMODWeb can also solicit
help from the email lists either at this site or at GMOD's homepage
(\url{http://gmod.org}) where a community of users is very active. 

\subsection{Unresolved Challenges}

% FIXME: could add info about future platforms supported with Turnkey
As with any open source development project, challenges remain for GMODWeb.
Although the project has been available for two years, it has only recently
released a 1.0 version.  It has been a challenge to attract new MOD users and
developers when the project was in this pre-release stage.  With the release of
ParameciumDB as a proof of concept for GMODWeb in a production environment, the
prospects for attracting both new users and developers have improved.  Another
challenge for the project is the very integrative nature of the GMODWeb
application.  Since it attempts to bring together several very large web
applications, the dependencies for the project are daunting.  Maintenance on
the various GMOD components integrated into GMODWeb has taken up a large
percentage of the development time.  However, as a beneficial side effect of
this effort, many useful GMOD tools have been packaged as RPMs for distribution
through the Biopackages repository, making them available for other projects
and uses.

\section{Figure Captions}

\subsection{Figure 6.1}

GMODWeb is the result of customizations to a Turnkey website built
with the Chado schema.  The GMODWeb skin was the product of modifications
mainly to the view layer.  This included changes to the template view layer including
overriding default templates and CSS changes. Enhancements were also performed
with layout changes through controller XML files modifications. 

\subsection{Figure 6.2}

Overviews of the Turnkey::Generate and Turnkey::Render processes.  A.
The process of creating a Turnkey-based website via Turnkey::Generate is shown.
A SQL schema file is processed using SQL::Translator to create a directed graph
representation of the relationships between tables. These are used by
Turnkey::Generate to create an MVC-based web application. B. The rendering of a
Turnkey page by Turnkey::Render is show.  When a client request is received an
XML document describing the relationships between objects is consulted.  Model
objects are created and combined with templates by the atom controller layer to
produce a rendered page. This is returned to the client. 

\subsection{Figure 6.3}

An example gene feature rendered with the customized ParameciumDB
skin.

\ifisthesis
\else
\section{Acknowledgments}

\emph{OA and LS acknowledge support from the CNRS and from ACI IMPBio2004 contract 14.}

% FIXME
\emph{USDA/ARS and NIH support GET FROM LINCOLN}
\fi

\newpage

% Figures
% FIXME: genome research wants figure legends then figures

\ifisthesis
\begin{figure}[h] \label{FIGURE 1} \includegraphics[width=11cm]{figures/figure1.pdf}
\else
\begin{figure}[h] \label{FIGURE 1} \includegraphics[width=11cm]{figure1.pdf}
\fi
\caption{GMODWeb and its relationship to Turnkey.}
\end{figure}

\ifisthesis
\begin{figure}[h] \label{FIGURE 2} \includegraphics[width=12cm]{figures/figure2.pdf}
\else
\begin{figure}[h] \label{FIGURE 2} \includegraphics[width=12cm]{figure2.pdf}
\fi
\caption{Overviews of the Turnkey::Generate and Turnkey::Render processes.} \end{figure}

\ifisthesis
\begin{figure}[h] \label{FIGURE 3} \includegraphics[width=14cm]{figures/figure3.pdf}
\else
\begin{figure}[h] \label{FIGURE 3} \includegraphics[width=16cm]{figure3.pdf}
\fi
\caption{An example gene feature rendered with the customized ParameciumDB
skin.}\end{figure}


\begin{table}
\label{TABLE 1}

\begin{scriptsize}
\caption[The GMODWeb application's software dependencies]{The GMODWeb application has many software dependencies.  This table
shows the immediate GMODWeb and Turnkey dependencies on the Fedora Core 2 Linux
distribution.  All dependencies are available from the Biopackages software
repository (http://biopackages.net) or are provided by the underlying operating
system.}

% FIXME: add a description field
% the following layout does not work with latex2rtf
\begin{tabular}{l|p{1cm}|p{1cm}|p{5cm}}
% this layout does work with latex2rtf
%\begin{tabular}{l|l|l|l}
\textbf{Package Name} & \textbf{Version} & \textbf{GMOD Tool} & \textbf{Description} \\
\hline
%& postgresql-libs &  \\
postgresql-server &  & & The PostgreSQL database server. \\
%& postgresql-devel  & yes \\
postgresql & $\geq$ 7.3 & &  Client libraries for PostgreSQL.\\
perl-Apache-ParseFormData & &  & A perl library for accessing form data in mod\_perl. \\
perl-Apache2-
%& perl-Apache-Session & & & \\
perl-Class-Base & & &  A perl base class for other modules. \\
perl-Class-DBI & & &  A perl tool for abstracting database access. \\
perl-Class-DBI-ConceptSearch & & &  A flexible perl module for searching databases. \\
perl-Class-DBI-Pager & & &  A perl tool for breaking database query results into pages. \\
perl-Class-DBI-Pg & & &  A PostgreSQL driver for Class::DBI. \\
perl-Class-DBI-Plugin-Type & & &  A perl tool for determining data type information. \\
perl-DBD-Pg & & &  A PostgreSQL driver for perl. \\
perl-DBI & & &  A generic database interface for perl. \\
perl-Log-Log4perl & & &  Logging software for perl applications. \\
perl-SQL-Translator & & & A perl tool for translating SQL schema into an object model. \\
perl-Template-Toolkit & & &  A template engine for perl. \\
perl-XML-LibXML & & &  An XML parsing library for perl. \\
perl-Lucene & & &  A Lucene search engine interface for perl. \\
perl-Apache2-SOAP & & &  Automatic SOAP bindings for mod\_perl. \\
perl-Cache-Cache & & &  A perl tool used to cache web pages in GMODWeb. \\
httpd & & &  The Apache webserver. \\
mod\_perl & $\geq$ 2.0.1 & &  A plugin for Apache that executes perl code. \\
perl & & & An interpreted language used throughout the Turnkey/GMODWeb project.\\
gbrowse & & yes &  A genome feature browser web application from the GMOD project. \\
textpresso & & yes &  A literature search web application from the GMOD project. \\
AmiGO & & yes &  An ontology browser web application from the GMOD project. \\
chado & & yes &  A sample Chado database from the GMOD project. \\
%& biopackages & & & yes & \\
chado-schema & & yes & The Chado schema from the GMOD project.\\
gmod-web & $\geq$ 1.3 & yes & A GMODWeb site generated with Turnkey for the Chado schema. \\
turnkey & $\geq$ 1.3 & & The website generation tool used to create GMODWeb. \\
\end{tabular}

\end{scriptsize}
\end{table}


% References
\ifisthesis
\bibliography{gmodweb_03-25-07} \bibliographystyle{plain}
\else
\bibliography{b} \bibliographystyle{genres}
\fi


