The first human genome sequence, completed in 2001, ushered in a new era of
biology.  This effort accelerated the development of biochemical assay and
information technologies.  As a result, biologists are now able to ask
questions that were previously considered intractable.  One example of a
breakthrough in assay technology resulting from genome sequencing efforts is
the DNA microarray, a high-throughput measurement device which enables
individual scientists to rapidly and simultaneously interrogate the RNA
concentration levels of virtually all genes in the human genome for a single
biological source.  The microarray is one of the most commonly used of
post-genomic high-throughput assay technologies, as is indicated by the number
of abstracts in PubMed containing the word ``microarray''.  It has been widely
adopted in the biological sciences and been used to address a large number of
research questions across a diverse set of subdomains in biological research.

Experiments that collect data using microarray technology can be divided into
several distinct categories, based on their overall goal and tactics employed
in analysis of the data.  These include: classification problems (supervised
learning) \cite{Dudoit2003ICM}, clustering problems (unsupervised learning)
\cite{Azuaje2003cge,Stanford2003cac}, gene co-regulation and function
identification studies \cite{PMID_12413821}, differential gene expression
studies \cite{PMID_15843092,PMID_15867208}, time-course and dose-response
studies \cite{PMID_12443997}, gene pathway and regulatory network studies
\cite{PMID_16216773,PMID_16825123}, drug discovery and toxicology studies
\cite{PMID_16512775,PMID_16700885,PMID_16880944,PMID_17195470}, clinical
diagnostics \cite{PMID_16918486}, and sequence-variation studies
\cite{PMID_17265721}.

A successful microarray experiment is one which is able to provide insight into
the hypothesis for which it was designed.  Insights gained by experimentation
are hard-earned, and require careful planning and execution of multiple key
steps.  The initial steps of feature selection, microarray design, microarray
fabrication, experiment design, and biological sample processing, and image
acquisition are described generally in Sections
\ref{ArrayDesign}-\ref{Hybridization}.  A more exhaustive discussion of
protocol surrounding these steps is given in \cite{wit2004}.  Following their
acquisition, images are quantified, stored, distributed, analyzed, and
analytical results shared.  Challenges and findings related to these latter
procedures are the subject of \emph{\dbthesis}.  They are thoroughly discussed
in Sections \ref{Data Processing}-\ref{Informatics}.  The entire process, from
microarray design to data analysis has been formally described as an object
model known as the Microarray Gene Expression Object Model (MAGE-OM)
\cite{mage}.  In this document I have borrowed vocabulary defined in the
MAGE-OM for clarity and consistency.  The MAGE-OM itself is discussed in
Section \ref{MAGE}.

% DONE
\section{DNA Microarray Design, Fabrication, and Assay Protocols}
\label{Assay}
\input{section_dna_microarray_assay}

% DONE
\section{DNA Microarray Experiment Design}
\label{Experiment Design}
\input{section_dna_microarray_experiments}

% DONE
\section{DNA Microarray Data Processing}
\label{Data Processing}
\input{section_dna_microarray_data_processing}

% WORKING
\section{DNA Microarray Informatics}
\label{Informatics}
\input{section_dna_microarray_informatics}

\section{Placeholder Section}
\label{XXX}

\bibliography{used}
%\bibliography {affy,scholar,pmid,thesis,la,dboutz}    % bibliography references
%\bibliographystyle {uclathes}
\bibliographystyle {plain}

