\documentclass{bioinfo}
\usepackage{algorithm2e}
\usepackage{graphicx}
\usepackage{color}
\usepackage{psfig}
\usepackage{times}
\usepackage{setspace}
\usepackage{url}
\copyrightyear{2008}
\pubyear{2008}


\begin{document}
\firstpage{1}

\title[Co-expression Analysis]{Gene Characterization Throught Large-Scale Co-expression Analysis}
\author[Day \textit{et~al}]{Allen Day\,\footnote{these authors contributed equally to this work}~ Jun Dong\,$^{*}$ and Stanley F. Nelson\,\footnote{to whom correspondence should be addressed}}
\address{Department of Human Genetics, David Geffen School of Medicine, University of California, Los Angeles.}
\history{Received on XXXXX; revised on XXXXX; accepted on XXXXX}
\editor{Associate Editor: XXXXXXX}
\maketitle

\begin{abstract}

%% provisional from stan

\section{Motivation:}
The major goal of human genetics is to identify normal variants and mutations
that lead to specific human phenotypes.  Linkage and association are powerful
tools to identify limited regions of the genome for further analysis to
identify DNA variants causative of disease.  However, the identification of
individual genes responsible for the phenotype is often elusive.  One way to
accelerate the identification of gene mutations and variations is to integrate
gene expression patterns, which provide an additional dimension of information.

\section{Results:}
We describe a broad-based tool that uses all publicly available gene expression
data on a single arraydesign to construct gene-gene expression correlations, and
can be used by individual scientists to explore the gene-gene co-expression
relationships present in this massive dataset to infer biological roles for
individual genes.  These matrices include virtually all human genes and
aggregate data from all laboratories depositing data on the web, as available
from Celsius, a community resource of image files and pre-processed Affymetrix
microarray data.

We show the power of this approach to prioritize sequencing of individual genes
within typical linkage intervals, and suggest novel candidate genes within
linkage regions for Joubert syndrome and Type 2 Limb girdle muscular dystrophy.
We demonstrate a technique that can reveal associations between genes and
biological processes, given a seed set of characterized genes, and present a
set of novel genes involved in cartilage development.

\section{Availability:}
All data used in these analyses, as well as web-based exploratory tools, are
available from \url{http://genome.ucla.edu/projects/XXX}.  Additionally,
software for programmatic access to these data is provided at the Bioconductor
website \url{http://bioconductor.org} in the XXX package.

\section{Contact:} \href{snelson@ucla.edu}{snelson@ucla.edu}
\end{abstract}

\section{Introduction}

Genetic linkage and association studies lead to the identification of links
between causal genes or genomic regions and their resulting phenotypes.  These
studies contain unique data, such as pedigree structures, that are not
obtainable through other methods of inquiry.  Data collected using DNA
microarrays also give a unique perspective on genes associated with a
particular phenotype, as the regulation of gene expression is independent of
genomic location.

As the genome era draws to a close, the number of uncharacterized diseases that
are influenced by a single or small number of genes has become very small.
This is because the statistical power required to identify causal links between
genes and phenotypes increases with the number of genes involved in the
biological processes that manifest in the phenotype.  Thus, a major theme in
the post-genome era is that advances will come through integrative approaches
that combine biological data from many sources \cite{liver,operon,csbdb}.

Previous efforts have demonstrated the utility of using gene-gene co-expression
patterns to perform gene characterization
\cite{webtom,carlson,aspm,wormnet,agca}.  Indeed, this type of retrospective
analysis of large volumes of data is characteristic of microarray studies in
general.  However, microarray studies frequently suffer from lack of
statistical power because of the relatively small numbers of samples observed
relative to the number of genes measured \cite{dimensionality}.  We addressed
this problem of dimension transposition simply by increasing the amount of data
in our analyses.  That is, we constructed Celsius, a publicly accessible data
warehouse of Affymetrix microarray data \cite{celsius}.  Celsius contains more
than 120,000 microarrays, and for the current-generation human arraydesigns
more than 12,000 arrays each.  Further, microarrrays added to Celsius are
processed in such a way to allow for progressive growth of the data set without
needing to reprocess data.

Celsius contains very little experimental metadata.  This position is in stark
contrast to the microarray community at large, which favors metadata collection
and modeling in the extreme \cite{mo,mage,miame}.  While we concede that
metadata provides additional statistical power in data analysis, policies that
require metadata to be deposited concommitantly with assay measurements may
actually be harmful.  By increasing barriers to public release of data via
metadata requirements, the amount of publicly available data is diminished.
This reduction in data volume then effectively reduces our ability to draw
biological conclusions from assay data alone.

Here, we demonstrate that fears of having vast volumes of unlabeled microarray
data are unfounded.  Quantitative microarray data are useful in their own
right, and can be used to address biological problems completely unrelated to
the initial study for which they were produced.

%Gene expression data represent an additional dimension of information that can
%be used to boost statistical power, thus leading to identification of more
%disease genes.  As with linkage and association data, the measurements made by
%whole-genome DNA microarrays provide unique information about gene-gene
%relationships, as the regulation of gene expression is independent of genomic
%location.

\section{Approach}
Text Text Text  Text Text Text Text Text Text Text Text

\begin{methods}
\section{Methods}

% <execute lang="bash"> wc -l
% /home/allenday/cvsroot/celsius/dump/data2/HG-U133_Plus_2.rma.sxe.col | awk
% '{print $1}' > m0_col.tex wc -l
% /home/allenday/cvsroot/celsius/dump/data2/HG-U133_Plus_2.rma.sxe.idx | awk
% '{print $1}' > m0_row.tex
% </execute>

\subsection{Data Processing}\label{Processing}

We retrieved RMA-processed gene expression data for the HG-U133\_Plus\_2
arraydesign from the Celsius microarray data warehouse \cite{rma,celsius} and
denote the $S$=\input{m0_row} array $\times$ $P=$\input{m0_col} probeset matrix
as $M$. %(Equation \ref{equation:M}).

%\begin{equation}\label{equation:M}
%M_{s,p} =
%\begin{pmatrix}
%m_{1,1} & \cdots & m_{1,p} \\
%\vdots  & \ddots & \vdots  \\
%m_{s,1} & \cdots & m_{s,p} \\
%\end{pmatrix}
%\end{equation}

A cursory examination of $M$ that there were abberant arrays present and that
these arrays would have a negative impact on any downstream analyses.  At a
coarse level, there appeared to be at least 3 types of aberrant arrays:

\renewcommand{\labelenumi}{\Alph{enumi}.}
\begin{enumerate}
\item arrays with extremely high gene expression values across many probesets.
\item arrays with extremely low gene expression values across many probesets.
\item arrays with dissimilar expression values for two probesets reputedly
measuring the same gene.
%\item arrays with dissimilar expression values for genes known to be
%co-expressed (Section \ref{corpass}).
\end{enumerate}

We sought to remove these abberant arrays from the dataset.

% <execute lang="R">
% z             = read.table('/home/allenday/cvsroot/celsius/dump/data2/HG-U133_Plus_2.rma.sxe.summary2');
% trim.fraction = 10;
% brightdim.cutoff = 3;
% trim.length   = as.integer(length(z[,5]) / trim.fraction);
% z5            = sort(z[,5]);
% z5.trim       = z5[ trim.length:(length(z5)-trim.length) ];
% postscript('array_means.ps',width=8,height=8,horizontal=FALSE,onefile=FALSE,paper="special",encoding="TeXtext.enc");
% hist(z5.trim,breaks=50,xlab="mean probeset intensity",ylab="frequency",main="mean probeset intensity of arrays");
% dev.off();
% write(trim.fraction, 'trim_fraction.tex');
% write(brightdim.cutoff, 'brightdim_cutoff.tex');
% write(mean(z5.trim),'trimmed_means_mean.tex');
% write(sd(z5.trim),'trimmed_means_sd.tex');
% write(sum( (z5 - mean(z5.trim)) / sd(z5.trim) >  brightdim.cutoff ),'bright_count.tex');
% write(sum( (z5 - mean(z5.trim)) / sd(z5.trim) < -brightdim.cutoff ),'dim_count.tex');
% write(sum( (z5 - mean(z5.trim)) / sd(z5.trim) < -brightdim.cutoff ) + sum( (z5 - mean(z5.trim)) / sd(z5.trim) >  brightdim.cutoff ),'brightdim_count.tex');
% write((1:length(z[,5]))[(z[,5] - mean(z5.trim)) / sd(z5.trim) > 3 | (z[,5] - mean(z5.trim)) / sd(z5.trim) < -3],'/home/allenday/cvsroot/celsius/dump/data2/HG-U133_Plus_2.rma.sxe.brightdim',ncolumns=1);
% </execute>
%% <execute lang="bash">
%% /home/allenday/cvsroot/celsius/dump/bin/i386-redhat-linux-gnu.4/trim.exe 54675 /home/allenday/cvsroot/celsius/dump/data2/HG-U133_Plus_2.rma.sxe.brightdim /home/allenday/cvsroot/celsius/dump/data2/HG-U133_Plus_2.rma.sxe.dat data2/HG-U133_Plus_2.rma_trim1.sxe.dat 
%% /home/allenday/cvsroot/celsius/dump/bin/trim_index.pl /home/allenday/cvsroot/celsius/dump/data2/HG-U133_Plus_2.rma.sxe.brightdim /home/allenday/cvsroot/celsius/dump/data2/HG-U133_Plus_2.rma.sxe.idx /home/allenday/cvsroot/celsius/dump/data2/HG-U133_Plus_2.rma_trim1.sxe.idx 
%% cp /home/allenday/cvsroot/celsius/dump/data2/HG-U133_Plus_2.rma.sxe.col /home/allenday/cvsroot/celsius/dump/data2/HG-U133_Plus_2.rma_trim1.sxe.col 
%% </execute>
% <execute lang="bash">
% wc -l /home/allenday/cvsroot/celsius/dump/data2/HG-U133_Plus_2.rma_trim1.sxe.col | awk '{print $1}' > m1_col.tex
% wc -l /home/allenday/cvsroot/celsius/dump/data2/HG-U133_Plus_2.rma_trim1.sxe.idx | awk '{print $1}' > m1_row.tex
% </execute>

\subsubsection{Removal of Dim and Bright Arrays}\label{dimbright}

Class A \& B arrays were easiest to identify.  Details of the exclusion
procedure are shown in Algorithm \ref{algorithm:mprime}.  Essentially, we
calculated the mean expression value of all probesets for each array, then
calculated the mean and standard deviation of a \input{trim_fraction}\% trimmed
distribution of those means.  The trimmed means themselves had a mean of
\input{trimmed_means_mean} and a standard deviation of
\input{trimmed_means_sd}.  A histogram of the distribution is given in Figure
\ref{figure:means}.  There were \input{brightdim_count} arrays with mean a
expression value more than \input{brightdim_cutoff} standard deviations away
from the mean of trimmed means.  These were primarily dim arrays
(n=\input{dim_count}) but there were also bright arrays
(n=\input{bright_count}).  These arrays were removed from further
consideration, leaving matrix $M\prime$ with \input{m1_row} arrays and
\input{m0_col} probesets.

\subsubsection{Removal of Inconsistent Arrays}\label{inconsistent}

Class C arrays were slightly more difficult to find.  To identify them, we
exploited the fact that, via NetAffx \cite{netaffx}, Affymetrix publishes a
probeset $\rightarrow$ gene symbol mapping for their array designs.  We assumed
that pairs of probesets designed to target the same gene were more likely to be
linearly related than randomly selected pairs because they were targetting the
same gene, and that these relationships could be used as a starting point to
identify inconsistent arrays.

We took all \input{symbol_count} unique gene symbols from the NetAffx
HG-U133\_Plus\_2 gene annotation, and identified the subset $G$
(n=\input{symbol_gt2_count}) for which there were two or more probesets.  We
constructed $G$ groups, each corresponding to a single gene symbol, i.e.
$g_1={p_{g_1}1,\dots,p_{g_1}n},\dots,g_G={p_{g_G}1,\dots,p_{g_G}n}$.  Then, for
each $g{\in}G$, we performed a linear regression of $log_{10}(signal)$ for all
possible probeset pairs $p_gA,p_gB$ (n=\input{symbolgroup_pair_count}).

Examination of the probeset pairs with the largest value of $r^2$ revealed that
the majority were control probesets that targeted spike-in sequences that are
added as part of the microarray hybridization for quality control.  Thus, we
concluded that using the built-in control probesets was a robust way to
identify aberrant arrays.  We performed \input{control_probesets_count}
multiple regressions, allowing each control probeset to be the response
variable once.  In the context of a single regression if an array's residual
was, relative to all other arrays' residuals, more than 3 standard deviations
away from the line, we incremented a counter for that array.  After performing
all \input{control_probesets_count} regressions, all arrays that were observed
more than 3 standard deviations more than 5\% of the time
(n=\input{outlier_badrow_count}) were removed from further consideration,
leaving a matrix $M\prime\prime$ with \input{m2_row} arrays and \input{m0_col}
probesets.  This procedure is expressed Algorithm \ref{algorithm:mprimeprime}.
Outlier frequencies per array are shown in Figure \ref{figure:rsquare}.

% <execute lang="bash">
% cat /home/allenday/cvsroot/celsius/dump/analysis2/HG-U133_Plus_2.symbol | awk '{print $1}' | sort | uniq | wc -l > symbol_count.tex
% cat /home/allenday/cvsroot/celsius/dump/analysis2/HG-U133_Plus_2.symbol | awk '{print $1}' | sort | uniq -c | grep -vw 1 | wc -l > symbol_gt2_count.tex
% cat /home/allenday/cvsroot/celsius/dump/analysis2/HG-U133_Plus_2.lm_rsquared | wc -l > symbolgroup_pair_count.tex
%% ./bin/i386-redhat-linux-gnu.4/transpose.exe 54675 ./data2/HG-U133_Plus_2.rma_trim1.sxe.dat ./data2/HG-U133_Plus_2.rma_trim1.exs.dat
%% ./bin/transpose_index.pl data2/HG-U133_Plus_2.rma_trim1.sxe.idx data2/HG-U133_Plus_2.rma_trim1.sxe.col data2/HG-U133_Plus_2.rma_trim1.exs.idx data2/HG-U133_Plus_2.rma_trim1.exs.col 
% </execute>

% <execute lang="R">
% #now build up the "repeat offender" vector
% source('/home/allenday/cvsroot/celsius/dump/lib/R/Celsius/IO/exs.R');
% conn = Celsius.exs('/home/allenday/cvsroot/celsius/dump/data2/HG-U133_Plus_2.rma_trim1');
% ctrl1=e_names(conn)[substr(e_names(conn), 1, 4)=="AFFX"];
% mat = get(conn, ctrl1, s_names(conn)); dim(mat); mat=t(mat); dim(mat);
% ### mat now has 12100 rows and 62 columns.
% J = length(ctrl1); # J is actually ncol(mat)
%
% #### The following considers 1 cutoff value ####
% residual.cutoff = 3;
% residual.tally  = rep(0, nrow(mat) ); r2.tally=rep(NA, J);
%
% for ( j in 1:J ) {
%   print(j);
%   mat.lm = lm(mat[,j] ~ mat[,-j]);
%   #the mean residual is always 0.
%   #mat.residual.sd = as.vector( ( residuals(mat.lm) - mean(residuals(mat.lm)) ) / sd(residuals(mat.lm)) );
%   mat.residual.sd = residuals(mat.lm) / sd(residuals(mat.lm));
%   residual.tally = residual.tally + (abs( mat.residual.sd ) > residual.cutoff);
%   r2.tally[j] = summary(mat.lm)$r.squared;
% }
% #rownames(residual.tally) = rownames(mat);
% table(residual.tally);
% sum(residual.tally > J*0.1 );
% # residual.tally = residual.tally / J;
% # rt3=residual.tally;
% postscript('probeset_residuals.ps',width=8,height=8,horizontal=FALSE,onefile=FALSE,paper="special",encoding="TeXtext.enc");
% plot(sort(residual.tally), xlab='arrays, sorted by r^2', ylab='outlier frequency', main='array outlier frequency');
% abline(5, 0, col='red');
% dev.off();
% #summary(r2.tally)
% r2.tally;
% write.table(ctrl1,'control_probesets.tex', row.names=FALSE, col.names=FALSE );
% write.table(residual.tally, "residual.tally.txt", sep="\t");
% write.table( (1:12100)[residual.tally>5] , 'outlier_badrow.tex', row.names=FALSE, col.names=FALSE );
% </execute>

% <execute lang="bash">
% cat control_probesets.tex | wc -l > control_probesets_count.tex
% wc -l /home/allenday/cvsroot/celsius/dump/data2/HG-U133_Plus_2.rma_trim2.sxe.idx | awk '{print $1}' > m2_row.tex
% cat outlier_badrow.tex | wc -l > outlier_badrow_count.tex
% </execute>

\subsubsection{Correlating Genes}\label{Correlation}

Subsequent to filtering out abberant arrays from out dataset (Section
\ref{Processing}), we used the $M\prime\prime$ matrix to calculate
$C\prime\prime$, a \input{m0_col} $\times$ \input{m0_col} matrix of Pearson
correlation coefficients for every pair of probesets (Equation
\ref{equation:cprimeprime}).  $C\prime\prime$ was used in all results presented
in Section \ref{Discussion}.

\begin{equation}\label{equation:cprimeprime}
C\prime\prime = cor({M^T}\prime\prime)
\end{equation}

\subsection{Annotating Genes}\label{Annotation Methods}

For each probeset $p \in P$ on the HG-U133\_Plus\_2 arraydesign, we retrieved
and sorted in descending order $r = C{\prime\prime}_p$.  We took $r\prime$, the
derivative of $r$, and used the R Bayesian Change Point \emph{bcp} to identify
$\delta$, the index of the largest value of $r\prime$ that preceded a
mostly-linear portion of the curve.
%A single best change point $\delta$ was
%selected as the index of the largest value of $r$ that preceded 10 consecutive
%values $\hat{p} < 0.5$. 
The subset of probesets where $r > \delta$ were defined as $Q$, and used as
input to the \emph{hyperGTest} function of the \emph{GOstats} package of
Bioconductor \cite{bioconductor} to test for enrichment of Gene Ontology (GO)
Biological Process (BP) annotations in a gene set.  \emph{hyperGTest} produced
a set of predicted gene annotations $N_p$ for each $p \in P$ based on the
annotation of neighbors $Q$.  We applied Bonferroni correction to the p-values
associated with each prediction by multiplying each p-value by the total number
of predictions made for the corresponding probeset.  We used these corrected
p-values from predicted annotations $N_p$ that were known to be
non-computationally assigned from the \emph{hgu133plus} package of Bioconductor
\cite{bioconductor} to establish a conservative cutoff, below which predicted
annotations should all be high-quality.  This process is presented in Algorithm
\ref{algorithm:annotation}.

\subsection{Analyzing Linkage Regions}\label{Linkage}

For a given phenotype, a group of known genes $G$ known to be associated with
that phenotype were retrieved from previous publications and online databases.
The list of genes was transformed to a list of probesets $P$ present on the
HG-U133\_Plus\_2 arraydesign using the gene symbol $\rightarrow$ probeset
mapping available from NetAffx \cite{netaffx}.  Probesets $P$ were then mapped
to 6-megabase genomic regions $A$ by finding the center point of each
probeset's alignment to UCSC's March 2006 (hg18) version of the human genome
and expanding by 3 megabases in each direction.  Each region in $A$ was then
mapped to a list of all HG-U133\_Plus\_2 probesets $Q$ aligned to that region.
Then, for each $p \in P$, a $Q{\times} G(G \ni g)$ slab was retrieved from
$C\prime\prime$ (Section \ref{Correlation}), and row-summarized to produce a
$Q$-length vector $\vec{\bar{r}}$ of mean correlation coefficients to $G \ni
g$.  The algorithm for this procedure is presented in Algorithm
\ref{algorithm:linkage}.

\end{methods}

\section{Discussion}\label{Discussion}

Our aim was to mine the matrix of correlation coefficients for all probesets on
the Affymetrix HG-U133\_Plus\_2 arraydesign for new information.

We wanted to let the data speak for themselves, and so included only a minimum
of metadata.  Metadata for samples hybridized to the arrays were excluded
entirely from analyses.  For probesets, we only included gene-symbol
\cite{netaffx}, genomic alignment \cite{karolchik}, and human-reviewed Gene
Ontology (GO) Biological Process (BP) \cite{go, bioconductor} metadata.

\subsection{Data Processing}\label{Processing Results}

All HG-U133\_Plus\_2 arrays (n=\input{m0_row}) were retrieved from Celsius
\cite{celsius}.  We assessed the arrays using some simple quality control (QC)
metrics, and excluded several hundred arrays as described in Section
\ref{Processing} yielding a \input{m2_row} array $\times$ \input{m0_col} column
matrix, denoted $M\prime\prime$.  We calculated the Pearson correlation
coefficient for every pair of probesets in $M\prime\prime$, yielding a
\input{m0_col} $\times$ \input{m0_col} correlation matrix, denoted
$C\prime\prime$.

\subsection{Disease Gene Recovery}\label{Recovery}

Usually the first published evidence of association between a hereditary
disease and one or more genes does not explicitly refer to the associated genes
but rather describe the association to multiple associated genetic loci that
should be examined more closely \cite{joubert,microcephaly}.  These so-called
linkage regions are commonly up to 10 megabases in size, and thus typically
contain 60-100 genes, assuming an average gene size of 50 kilobases.

When the associated genes are eventually identified, it is frequently the case
that they are all involved in the same biological process, and that this
process is disrupted when one of its components is dysfunctional.  Given that
the genes are involved in the same biological process, it is reasonable to
assume that they will be coexpressed in cells where the process occurs and
thus be positively correlated.

Extending the idea that genes involved in the same biological process will
generally be positively correlated, we sought to use $C\prime\prime$ (Section
\ref{Processing Results}) to simulate the identification of a disease gene.

Our method was to assemble a list of genes $G$ known to be associated with a
disease.  Each gene identifier $g \in G$ was mapped to the corresponding list
of probesets on the HG-U133\_Plus\_2 arraydesign.  The list is denoted $P_g$,
and is derived from the mapping function denoted $J(g)$.  For each probeset in
$p_g \in P_g$, the genomic position was retrieved using the UCSC Genome
Browser, human build hg18 (March 2006) \cite{karolchik}.  We then retrieved a
list of probesets which aligned to a 6 megabase genomic region surrounding the
initial probeset.  The list is denoted $Q_{p_g}$, and is derived form the
mapping function denoted as $K(p)$.  Next, the vector of mean correlaton
coefficient $\bar{r}_{p_g}$ of probeset $q_{p_g} \in Q_{p_g}$ to $P \ni J(g)$
was calculated using function $L(q)$ from $C\prime\prime$.  Finally, the best
gene in the region was identified as the one matching
$J^{-1}(K^{-1}(L^{-1}(max(\bar{r}_{p_g}))))$, the maximum value of $\bar{r}$
for named genes in the simulated linkage region.  If the gene identified was
present in $G$ we evaluated the result as positive.  In the event that genes
not in $G$ met the criterion of $max(\bar{r})$, we evaluated the result as
negative.

We first applied our method to the limb-girdle musscular dystrophy, type 2
(LMGD2) phenotype.  There are 11 genes known to be associated with LMGD2.  For
each gene, we considered all probesets targeting a named gene within a
6-megabase genomic region centered at the gene's locus.  We calculated the mean
correlation coefficient $\bar{r}$ to the 11-gene LMGD2 profile for each
probeset within the region, but excluding any probesets targeting the gene used
to select the region.  In 55\% (6/11) cases $\max{\bar{r}}$ corresponded to the
causal gene for LMGD2.

Next, we applied this method to see if we could identify the four genes known
to be associated with microcephaly.  The purpose of this test was to confirm
that the methods used in our LGMD2 trial would effectively identify genes for a
completely different phenotype, as well as to see if the method was robust
enough to identify the known gene given a much smaller profile for comparison.
In 75\% (3/4) of cases the most correlated gene with the other known
microcephaly genes was correctly identified from a 6-megabase linkage region
surrounding that gene.

Finally, we applied our scanning method to Joubert syndrome.  Seven linkage
regions for Joubert syndrome have been identified (JBTS1-JBTS7).  Five of
these have had the associated gene in the region identified (JBTS3=AHI1;
JBTS4=NPHP1; JBTS5=CEP290; JBTS6=TMEM67; JBTS7=RPGRIP1L)
\cite{JBTS3,JBTS4,JBTS5,JBTS6,JBTS7} while regions JBTS1 and JBTS2 have so far
only been linked to D9S158 \cite{JBTS1} on chromosome 9 and a 17-megabase
centromeric region of chromosome 11 \cite{JBTS2A,JBTS2B,JBTS2C}, respectively.

The purpose of this third test was another instance of reproducibility of
results, as well as to see if we could make a prediction as to the identity of
the genes in remaining linked regions JBTS1 and JBTS2 for which a gene has not
yet been identified.  We were able to correctly identify 80\% (4/5) of the five
genes known to be associated with Joubert syndrome.  A plot of $r$-values
surrounding NPHP1 is given in Figure \ref{figure:NPHP1}.  We also show data from the
Gene Expression Atlas \cite{GNF} for the same region in Figure \ref{figure:GeneAtlas}
to demonstrate that NPHP1 could not be identified merely by scanning this
region for brain-specific or even brain-expressed genes.

Based solely on the correlation data, we are able to suggest the
uncharacterized gene C9orf116 as the most likely candidate for the 6-megabase
region surrounding D9S158 that is synonymous with JBTS1 (Figure
\ref{figure:JBTS1}).

We also examined JBTS2 to see if we could suggest any genes that might be
associated with Joubert syndrome in this region as well.  JBTS2 is a
centromere-spanning 17-megabase region on chromosome 11 between markers
D11S1915 and D11S4191 (Figure \ref{figure:JBTS2}).  We included an additional
3-megabases upstream and downstream of the outer markers.  The best candidate
for JBTS2 is AGBL2.  However, this is a large region and there are
highly-correlated genes on both sides of the centromere.  For these reasons we
suggest that the 17-megabase centromere-spanning linkage region JBTS2 is
actually two separate linkage regions which we designate as JBTS2p and JBTS2q
located on the p- and q-side of the centromere, respectively.  The best
candidate in JBTS2p is AGBL2.  Additional candidates in JBTS2p are C11orf49 and
probeset 229687\_s\_at.  The best candidate in region JBTS2q is M4A8B.
Additionally candidates in JBTS2q are SCGB1A1 and probeset 229688\_at.

%joubert
% JBTS1                 9q34.3          unknown gene    D9S158
% JBTS2 aka CORS2       11p12-11q13.3   unknown gene    D11S1915(lod=5.2),D11S1313(lod=3.96),D11S4191-D11S1344(lod=NA)
% JBTS3                 6q23            AHI1
% JBTS4                 2q13            NPHP1
% JBTS5                 12q21.32        CEP290 (aka NPHP6)
% JBTS6                 8q21            TMEM67
% JBTS7                 16q12.2         RPGRIP1L

% D9S158   chr9  138138869 138339251
% D11S1915 chr11 45465651 45666307
% D11S1313 chr11 55890635 56091023
% D11S4191 chr11 59656135 59856421
% D11S1344 chr11 46023436 46223802
% union    chr11 42465651 62856421

%  We started with a set of 56 probesets
%representing 48 genes that were identified in a microarray study that searched
%for genes associated with skeletal abnormality \cite{funari}.  We used the UCSC
%Genome Browser \cite{karolchik} to align these probesets to the March 2006
%build of the human genome.

\subsection{Gene Functional Assessment}

Next, we considered whether these correlation data are more generally useful.
In the case of identifying disease genes, the search space was constrained to a
linkage region.  However, some analytical methods are position-independent and
seek to characterize all genes.  One such task is the assignment of functional
annotation.

To evaluate the performance of these correlation data for the assignment of
gene annotation into biological processes, we selected 5 genes that were
non-computationally annotated for the Gene Ontology term for ``muscle
contraction'' (GO:0006936).  The genes selected were UTRN, KCNQ1, MYOM1, SGCA,
and ASPH.  These are represented by 13 probesets on the HG-U133\_Plus\_2
arraydesign.  Next, we calculated $\bar{r}$, the mean correlation coefficient
to the 13-probeset profile, as described in Section \ref{Recovery}, but did so
genome-wide rather than restricting to a single linkage region.  Then, we
rank-ordered the set of all probesets and the set of the initially selected 13
probesets by $\bar{r}$.  We observed that the muscle probesets were generally
more highly-correlated to the profile, presumably because of the internal
consistency of the set.  However a large number of genes not used present in
the profile also had high values of $\bar{r}$, and we sought to evaluate how
well the 5-gene profile was able to recover other genes known to be involved in
muscle contraction.  To do so, we tested for enrichment of muscle contraction
annotations for the probesets in the top decile of $\bar{r}$ values,
partitioned into one-tenth percentile sized quantiles.  We observed that
enrichment for muscle contraction annotation is strongly correlated to an
increased value of $\bar{r}$ relative to our 5-gene profile.  The value of
$\bar{r}$ versus the cumulative distribution of probesets is given in Figure
\ref{figure:muscle_cdf}, along with the significance of the enrichment of muscle
contraction annotation within each quantile.

We then applied this method to the results of a previous study by Funari,
\emph{et al.} that identified several previously unannotated genes that are
expressed in cartilage tissue \cite{funari}.  This study identified 114 genes
represented by 133 probesets from the HG-U133A arraydesign that were highly
expressed only in cartilage (Table 1).  We performed the scan described, and
were able to positively identify the correct gene 66\% (75/114) times.  A
graphical representation of the 6-megabase region surrounding one cartilage
gene, COL9A2, is given in Figure \ref{figure:COL9A2}.

%\subsection{Gene Annotation}\label{Annotation Results}
%
%We observed that each probeset $p \in P$ on the HG-U133\_Plus\_2 arraydesign
%typically has only a few highly correlated neighbor probesets, while the
%majority of probesets are slightly correlated to slightly anti-correlated.  An
%example of the rapid drop in correlation coefficients for a single probeset
%1316\_at is shown in Figure \ref{figure:cortip}.  We hypothesized that the
%nearest neighbors were likely to be involved in the the same biological
%process, and that the observed transcriptional co-regulation would allow us to
%suggest roles for unannotated genes, and to suggest previously unknown roles
%for characterized genes.
%
%To this end, we first evaluated how well the co-regulated neighbors could be
%used to reconstruct human-curated annotation that were already assigned to the
%gene using the Biological Process (BP) aspect of the Gene Ontology (GO)
%\cite{go}.  To do so for a single BP-annotated probeset, we first needed to
%discriminate between nearby and distant neighbors and did so using the
%\emph{bcp}, the Bayesian Change Point library of the R programming language.
%The subset of nearby neighbors for which BP was available was then tested for
%category enrichment using a hypergeometric-based test provided by the
%Bioconductor \cite{bioconductor} package \emph{GOstats}.  We tested for
%enrichment of all GO BP categories associated with any of the nearby neighbors,
%and so adjusted all p-values output by the hypergeometric test using Bonferroni
%correction in order to compensate for multiple hypothesis testing.
%
%We then used the BP annotations that were \emph{already known} from existing
%literature and ``recovered'' solely based on nearby neighbor annotations to
%estimate the p-value below which novel predictions should be accurate.
%
%We were able to recover the correct annotation for XXX\% of known annotations.
%The relationship between fractional recovery and confidence of the prediction
%(p-value) is given in Figure XXX.  The vast majority of recovered annotations
%(XXX\%) had an adjusted p-value of less than XXX, and we used this as a cutoff
%below which predicted annotations should be accurate.
%
%More than XXX about 10,000 XXX probesets on the HG-U133\_Plus\_2 arraydesign are not
%assigned to any existing gene symbol.  This typically means they were designed
%to measure a transcript that is computationally predicted only, and not
%supported by any \emph{in vivo} observation.  An additional XXX about
%5,000-10,000 XXX probesets are assigned to a gene symbol that are supported by
%little or no \emph{in vivo} data.  This group contains predicted genes, as well
%as transcripts that have been observed in EST libraries.
%
%Typical probesets in both of these groups have no annotation whatsoever.  We
%were able to assign XXX annotations to XXX probesets (XXX\%) in these groups.
%The data provided here represent an initial, and thus significant, step forward
%in the characterization of the roles of these hypothetical and rarely observed
%genes.
%
%Our assignment of an additional XXX annotations to genes which are already
%characterized is also significant, as it suggests direct linkage between
%biological processes previously known to be only indirectly related.

\section{Conclusion}

These correlation data are generally useful, and may be used to address a wide
variety of biological questions.  We have demonstrated that we are able to
correctly identify causal genes within their linkage regions for several
unrelated disease phenotypes.  We also demonstrated the ability use these data
to extrapolate what is known about a disease process to prioritize candidates
within linkage regions for which the causal gene has not yet been identified.

Further, these correlation data are useful for characterizing non-disease
phenotypes and can be used to characterize genes at global-scale as opposed to
within specific genomic regions.  We demonstrated this using a small set of
muscle contraction genes to identify genes that are known to also be involved
in muscle contraction according to the Gene Ontology Consortium.  We reproduced
this result and demonstrated that this type of enrichment analysis is not
process- or annotation-specific by using a set of cartilage development genes
identified from a previous microarray study to find more genes likely to be
involved in cartilage development.

The versatility of this resource initially surprised us.  The methods used to
assemble the correlation matrix is completely metadata independent -- only the
genomic alignment of probe sequences and the quantitative measurements made by
the microarray were used.  The data set is also very heterogenous.  It is
composed from microarray data generated from thousands of individual
experiments, from hundreds of experimenters, with each experiment using
different biological materials and variations of the Affymetrix protocols.

What conclusions can we draw? XXX ... XXX

 
These results are in are somewhat surprising, but speak to the power...

\section*{Funding}
Text Text Text Text Text Text

\section*{Acknowledgement}
Text Text Text Text Text Text  Text Text.

\bibliography {b}
\bibliographystyle {plain}
\newpage
%\begin{thebibliography}{}
%\bibitem[Bofelli {\it et~al}., 2000]{Boffelli03} Bofelli,F., Name2, Name3 (2003) Article title, {\it Journal Name}, {\bf 199}, 133-154.
%\bibitem[Bag {\it et~al}., 2001]{Bag01} Bag,M., Name2, Name3 (2001) Article title, {\it Journal Name}, {\bf 99}, 33-54.
%\end{thebibliography}

%%%%%%%%%%%%%%%%
% figures
%%%%%%%%%%%%%%%%

% Plotted inline with text

\begin{figure}[!tpb]
\label{figure:means}
\centerline{\resizebox{9cm}{!}{\psfig{figure=array_means.ps}}}
\caption{Mean probeset signal intensity of arrays.  The mean value for all probesets was
calculated for each array (x-axis) and the frequency of any given mean is
plotted by bin (y-axis).  Several extremely bright arrays (n=15) are not shown.}
\end{figure}

% Plotted inline with text

\begin{figure}[!tpb]
\label{figure:rsquare}
\centerline{\resizebox{9cm}{!}{\psfig{figure=probeset_residuals.ps}}}
\caption{Regressions of control probesets reveal aberrant arrays.  Multiple regressions
were performed for all 62 HG-U133\_Plus\_2 control probesets.  Arrays (x-axis)
are plotted versus the fraction of observations with regression residual
$>3\sigma$ (y-axis).  A red horizontal line indicates a cutoff above which
arrays are omitted from analysis.}
\end{figure}

% <execute lang="R">
% center=TRUE;
% directory='limbgirdle2';
% gene='TTN';
% prefix='cart';
% postscript('TTN.ps',width=8,height=8,horizontal=FALSE,onefile=FALSE,paper="special",encoding="TeXtext.enc");
% source('./cart/regionplot.R');
% dev.off();
% </execute>

\begin{figure}[!tpb]
\label{figure:TTN}
\centerline{\resizebox{9cm}{!}{\psfig{figure=TTN.ps}}}
\caption{Gene correlations to a list of LMGD2-associated genes within a
6-megabase region surrounding the location of an associated gene.  The genomic
position (x-axis) of probesets within a 6-megabase region centered at the
location of a TTN, a gene known to be associated with LMGD2, is plotted versus
the Pearson correlation coefficient $r$ (y-axis) to a list of probesets
targetting other genes known to be associated with Joubert syndrome across
11636 HG-U133\_Plus\_2 microarrays.  Solid circles: probesets targeting TTN,
$\times$s: probesets not designed to target a known gene, open circles: other
probesets.}
\end{figure}

% <execute lang="R">
% center=TRUE;
% directory='cartilage';
% gene='COL9A2';
% prefix='cart';
% postscript('COL9A2.ps',width=8,height=8,horizontal=FALSE,onefile=FALSE,paper="special",encoding="TeXtext.enc");
% source('./cart/regionplot.R');
% dev.off();
% </execute>

\begin{figure}[!tpb]
\label{figure:COL9A2}
\centerline{\resizebox{9cm}{!}{\psfig{figure=COL9A2.ps}}}
\caption{Gene correlations to a cartilage-specific gene list within a 6-megabase region
surrounding the location of an associated gene.  The genomic position (x-axis)
of probesets within a 6-megabase region centered at the location of COL9A2, a
gene known to be associated with cartilage development, is plotted versus the
Pearson correlation coefficient $r$ (y-axis) to a list of cartilage-expressed
probesets targetting other genes across 11636 HG-U133\_Plus\_2 microarrays.
Solid circles: probesets targeting COL9A2, open triangles: other probesets in
the cartilage-expressed list, $\times$s: probesets not designed to target a
known gene, open circles: other probesets.}
\end{figure}

% <execute lang="R">
% center=TRUE;
% directory='joubert';
% gene='NPHP1';
% prefix='cart';
% postscript('NPHP1.ps',width=8,height=8,horizontal=FALSE,onefile=FALSE,paper="special",encoding="TeXtext.enc");
% source('./cart/regionplot.R');
% dev.off();
% </execute>

\begin{figure}[!tpb]
\label{figure:NPHP1}
\centerline{\resizebox{9cm}{!}{\psfig{figure=NPHP1.ps}}}
\caption{Gene correlations to a list of Joubert syndrome-associated genes within a
6-megabase region surrounding the location of an associated gene.  The genomic
position (x-axis) of probesets within a 6-megabase region centered at the
location of a NPHP1, a gene known to be associated with Joubert syndrome, is
plotted versus the Pearson correlation coefficient $r$ (y-axis) to a list of
probesets targetting other genes known to be associated with Joubert syndrome
across 11636 HG-U133\_Plus\_2 microarrays.  Solid circles: probesets targeting
NPHP1, open triangles: other probesets in the Joubert syndrome gene list,
$\times$s: probesets not designed to target a known gene, open circles: other
probesets.}
\end{figure}

% XXX do this

\begin{figure}[!tpb]
\label{figure:GeneAtlas}
\centerline{\resizebox{9cm}{!}{\psfig{figure=NPHP1_GeneAtlas.ps}}}
\caption{Normal tissue expression surrounding NPHP1.  Probeset position in
ascending genomic order (x-axis) versus tissue (y-axis) from the GNF Expression
Atlas 2 are presented as a tissue-clustered, column-scaled heatmap.  Black=low
expression, white=high expression.  Black bars in the margin indicate brain
tissue rows, and the column representing Joubert syndrome-associated gene
NPHP1.}
\end{figure}

% <execute lang="R">
% prefix='novel';
% center=FALSE;
% directory='joubert';
% gene='JBTS1';
% prefix='novel';
% postscript('JBTS1.ps',width=8,height=8,horizontal=FALSE,onefile=FALSE,paper="special",encoding="TeXtext.enc");
% source('./cart/regionplot.R');
% dev.off();
% </execute>

\begin{figure}[!tpb]
\label{figure:JBTS1}
\centerline{\resizebox{9cm}{!}{\psfig{figure=JBTS1.ps}}}
\caption{Gene correlations to a list of Joubert syndrome-associated genes within 3
megabases of JBTS1 marker D9S158.  The genomic position (x-axis) of probesets
within a 6-megabase region on chromosome 9 centered at the location of marker D9S158, the peak
of Joubert syndrome associated region JBTS1, is plotted versus the Pearson
correlation coefficient $r$ (y-axis) to a list of probesets targetting other
genes known to be associated with Joubert syndrome across 11636
HG-U133\_Plus\_2 microarrays.  $\times$s: probesets not designed to target a
known gene, open circles: other probesets.  The probesets located at ~137.5
megabases for C9orf116 are the best candidate for a Joubert syndrome gene
within this region.}
\end{figure}

% <execute lang="R">
% prefix='novel';
% center=FALSE;
% directory='joubert';
% gene='JBTS2';
% prefix='novel';
% postscript('JBTS2.ps',width=8,height=8,horizontal=FALSE,onefile=FALSE,paper="special",encoding="TeXtext.enc");
% source('./cart/regionplot.R');
% abline(v=(45465651/1000000),col='red');
% abline(v=(45666307/1000000),col='red');
% abline(v=(55890635/1000000),col='orange');
% abline(v=(56091023/1000000),col='orange');
% abline(v=(59656135/1000000),col='blue');
% abline(v=(59856421/1000000),col='blue');
% abline(v=(46023436/1000000),col='cyan');
% abline(v=(46223802/1000000),col='cyan');
% dev.off();
% </execute>

\begin{figure}[!tpb]
\label{figure:JBTS2}
\centerline{\resizebox{9cm}{!}{\psfig{figure=JBTS2.ps}}}
\caption{Gene correlations to a list of Joubert syndrome-associated genes within 3
megabases of the 17-megabase region JBTS2.  The genomic position (x-axis) of
probesets within a 26-megabase region centered at JBTS2, a centromere-spanning
17-megabase region of chromosome 11 demarcated by D11S1915 and D11S4191 known
to be associated with Joubert syndrome, is plotted versus the Pearson
correlation coefficient $r$ (y-axis) to a list of probesets targetting other
genes known to be associated with Joubert syndrome across 11636
HG-U133\_Plus\_2 microarrays.  $\times$s: probesets not designed to target a
known gene, open circles: other probesets.  Paired vertical lines indicate the
posiiton of markers used in the mapping of this region.  From left to right:
D11S1915 (red), D11S1344 (cyan), D11S1313 (orange), D11S4191 (blue).  The
probeset located at ~47 megabases for AGBL2 is the best candidate for a Joubert
syndrome gene within this region.}
\end{figure}

% <execute lang="R">
% .libPaths('/home/allenday/lib/R');
% library('bcp');
% source('/home/allenday/cvsroot/celsius/dump/lib/R/Celsius/IO/exe.R');
% conn.exe = Celsius.exe('/home/allenday/cvsroot/celsius/dump/data2/HG-U133_Plus_2.rma_trim2');
% d0 = sort(as.vector(get(conn.exe,e_names(conn.exe)[1])),decreasing=TRUE)[1:100];
% d1 = d0[1:99] - d0[2:100];
% d2 = (bcp(d1)$posterior.prob)[1:98];
% postscript('cortip.ps',width=8,height=8,horizontal=FALSE,onefile=FALSE,paper="special",encoding="TeXtext.enc");
% plot(d0[2:100],ylim=c(0,1),pch=4,xlab='neighbors, ordered by r',ylab='correlation r / changepoint probability of r\'',main=paste('identification of neighbors of ',e_names(conn.exe)[1]));
% lines(d2[1:98],lty=1);
% abline(v=9.5,lty=2,col='red');
% dev.off();
% </execute>

\begin{figure}[!tpb]
\label{figure:cortip}
\centerline{\resizebox{9cm}{!}{\psfig{figure=cortip.ps}}}
\caption{Identification of most correlated neighbors for probeset 1007\_s\_at.
Pearson correlation coefficient $r$ (y-axis) is plotted for the 100 most highly
correlated probesets to 1007\_s\_at (x-axis) using $\times$s.  Solid lines
indicates the probability of a change in slope of $r\prime$.  A vertical dotted
line indicates the position above which slope changes of $r\prime$ are common.}
\end{figure}

% <execute lang="R">
% source('/home/allenday/cvsroot/celsius/dump/lib/R/Celsius/IO/exe.R');
% conn.exe = Celsius.exe('/home/allenday/cvsroot/celsius/dump/data2/HG-U133_Plus_2.rma_trim2');
% know = read.table('/home/allenday/dissertation/allenday-dissertation/cor/cart/cartilage/cart.gene.txt');
% prof = get(conn.exe, know[,2]);
% prof.mean = apply(prof, 2, mean);
% know.frag = sort( as.vector(apply( prof[ , match(know[,2], colnames(prof)) ], 2, mean )) );
% postscript('cartilage_cdf.ps',width=8,height=8,horizontal=FALSE,onefile=FALSE,paper="special",encoding="TeXtext.enc");
% plot( sort(prof.mean), 1:length(prof.mean), axes=F, xlab='r', ylab='fraction of probesets', main='CDF of cartilage and all probesets Pearson r', type="l" );
% box();
% lines( know.frag, 1:length(know.frag) * (length(prof.mean)/length(know.frag)), lty=2 );
% axis(1, seq(-0.07,0.17, by=0.01 ));
% axis(2, 1:10 * length(prof.mean)/10, 1:10/10);
% dev.off();
% write.csv( cbind( as.numeric(as.logical(match(colnames(prof[ , prof.mean >= quantile(prof.mean, 0.995) ]), rownames(prof)))), prof.mean[ prof.mean >= quantile(prof.mean, 0.995) ], annAll[ match( colnames(prof[ , prof.mean >= quantile(prof.mean, 0.995) ]), annAll[,1] ), 1:3 ] ), 'cartilage_candidates.csv' )
% </execute>

\begin{figure}[!tpb]
\label{figure:cartilage_cdf}
\centerline{\resizebox{9cm}{!}{\psfig{figure=cartilage_cdf.ps}}}
\caption{What is plotted is the mean correlation coefficient to a cartilage profile of
each probeset known to be cartilage-expressed (red, x-axis) vs. the total
fraction of cartilage-expressed probesets <= a given value of r.  A CDF,
effectively.  The same is plotted for all probesets (black).  The purpose is to
show that there are some probesets in the black curve that are not already
known to be cartilage-expressed, i.e. they are not in the red curve.  But these
black points have higher correlation coefficient to the profile than some
fraction of red points, so they are likely to be cartilage expressed.  So is it
clear?  Is there a better way to do it?}
\end{figure}

% as.list(GOBPOFFSPRING)[['GO:0006936']]
% zcat ~/public_html/data/HG-U133_Plus_2.GO.known.dat.gz | grep -E '0006936|0002086|0006937|0006939|0006940|0006941|0006942|0030049|0031443|0031444|0031445|0031446|0031447|0031448|0031449|0031450|0031451|0031452|0031453|0031454|0043133|0043134|0045932|0045933|0045986|0045987|0045988|0045989' | grep -v IEA | wc

% <execute lang="R">
%# ourProbeset = scan('/clustertmp0/allenday/ourProbeset.dat',what='character');
%# ourUniverse = scan('/clustertmp0/allenday/ourUniverse.dat');
% source('/home/allenday/cvsroot/celsius/dump/lib/R/Celsius/IO/exe.R');
% conn.exe = Celsius.exe('/home/allenday/cvsroot/celsius/dump/data2/HG-U133_Plus_2.rma_trim2');
% know = read.table('/home/allenday/dissertation/allenday-dissertation/cor/cart/muscletest/cart.gene.txt');
% prof = get(conn.exe, know[,2]);
% know.frag = sort( as.vector(apply( prof[ , match(know[,2], colnames(prof)) ], 2, mean )) );
% prof.mean = apply(prof, 2, mean);
% jj = 901:1000/1000;
%# hresults = matrix(rep(NA,300),nrow=100);
% hresults = read.csv('muscle_contraction_p.csv');
%# for (j in 100:98) {overP = colnames(prof)[ prof.mean > quantile(prof.mean, jj[j]-0.01) & prof.mean <= quantile(prof.mean, jj[j]) ];overQ = ourUniverse[match(overP,ourProbeset)];params = new("GOHyperGParams", geneIds = overQ,  universeGeneIds = ourUniverse, annotation = "hgu133plus2", ontology = "BP", pvalueCutoff = 1.0, conditional = FALSE,  testDirection = "over");hgOver = hyperGTest(params);hresults[j,1] = jj[j]-0.01; hresults[j,2] = jj[j]; dd = summary(hgOver)[ summary(hgOver)$GOBPID == 'GO:0006936', 2 ]; print(paste(j,jj[j],hresults[j,1],hresults[j,2],dd));if (sum(dd)>0){ hresults[j,3]=dd; } else { hresults[j,3]=1; } }
% postscript('muscle_cdf.ps',width=8,height=8,horizontal=FALSE,onefile=FALSE,paper="special",encoding="TeXtext.enc");
% par(mar=c(5,8,4,5)+0.1);
% prof.mean.part = prof.mean[ prof.mean > quantile(prof.mean,0.95) ];
% zz = cbind( as.character(apply( prof[ , match(know[,2], colnames(prof)) ], 2, mean )), as.character(know[,1]) );
% plot( sort(prof.mean.part), 1:length(prof.mean.part), axes=F, xlab='r', ylab='fraction of probesets', main='', cex=0.5, type='l' ); box();
% points( know.frag, 1:length(know.frag) * (length(prof.mean.part)/length(know.frag)), pch=zz[ rev(rank(zz[,1])), 2 ] );
% lines(quantile(prof.mean,hresults[,3]),length(prof.mean.part)*-log10(hresults[,4])/max(-log10(hresults[,4])),lwd=1, lty=2);
% axis(1, seq(0.06,0.19, by=0.01 ));
% axis(2, 0:5 * length(prof.mean.part)/5, 95:100/100);
% axis(2, 0:10 * length(prof.mean.part)/10, 0:10/10,line=5);
% mtext("annotation -log10(p-value)", side=4, line=3);
% axis(4, 0:11 * length(prof.mean.part)/11, 4*0:11);
% dev.off();
% </execute>

\begin{figure}[!tpb]
\label{figure:muscle_cdf}
\centerline{\resizebox{9cm}{!}{\psfig{figure=muscle_cdf.ps}}}
\caption{Plot of $\bar{r}$ to a muscle profile of 5 genes (UTRN, KCNQ1, MYOM1,
SGCA, ASPH; 13 probesets) annotated with ``muscle contraction''.  The 13
probesets present in the profile (red, x-axis) and all probesets (black,
x-axis) are plotted versus the cumulative fraction of cartilage-expressed
probesets <= a given value of $\bar{r}$.  Hypergeometric test enrichment of
term ``muscle contraction'' $-log_10(p-values)$ within each 0.1\%-sized
quantiles for the top decile are also shown (cyan, right x-axis).}
\end{figure}



\begin{algorithm}\label{algorithm:mprime}
\SetLine
\KwIn{List of arrays}
\KwOut{List of arrays of typical brightness}

$m = []$\;
$good = []$\;
\ForEach{Array $s$}{
  $m = [m, mean(M_s)]$\;
}
$m_{trim} = sort(m)_{m*0.1} \dots sort(m)_{m*0.9}$\;
$\mu    = mean(m_{trim})$\;
$\sigma = standard_deviation(m_{trim})$\;
\ForEach{Array $s$}{
  \If{${\mid}mean(M_s) - \mu{\mid} / \sigma < 3$}{
    $good = [good,s]$\;
  }
}
\Return{good}\;
\caption{Identification and removal of dim and bright arrays}
\end{algorithm}

\begin{algorithm}\label{algorithm:mprimeprime}
\SetLine
\KwIn{List of arrays, List of control probesets}
\KwOut{List of arrays with consistent control probesets}

$mark = []$\;
$good = []$\;
\ForEach{Control probeset $c \in C$}{
  $lm = regression(M_{C \ni c}, response=M_c)$\;
  $r = residuals(lm)$\;
  $\mu = mean(r)$\;
  $\sigma = standard_deviation(r)$\;
  \ForEach{Array z-score $z = (r - \mu) / \sigma$}{
    \If{${\mid}z{\mid} > 3$}{
      $mark[z]++$\;
    }
  }
}
\ForEach{Array $a$}{
  \If{$mark[a] / length(C) < 0.05 $}{
    $good = [good, a]$\;
  }
}
\Return{good}\;
\caption{Identification and removal of arrays with deviant control probeset signals}
\end{algorithm}

\begin{algorithm}\label{algorithm:linkage}
\SetLine
\KwIn{$E$, a list of ``profile'' gene symbols}
\KwOut{$F$, a list of ``candidate'' probesets / gene symbols}
$P = []$\;
$best = []$\;
$hit = []$\;
\ForEach{Gene symbol $g \in G$}{
  $P = [P,probesets(g)]$\;
}
\ForEach{Gene symbol $g \in G$}{
  $b = genomic\_position(g)$\;
  $b_{min} = b - 3{\times}10^6$\;
  $b_{max} = b + 3{\times}10^6$\;
  $Q = probesets\_in\_region(b_{min},b_{max})$\;
  $best[g] = -1$\;
  \ForEach{Probeset $q \in Q$}{
    $T = probesets(E) \ni probesets(g)$\;
    $r = { {\sum_{t}^{T} C{\prime\prime}_t} \over {length(T)} }$\;
    \If{$r > best[g]$}{
      $bet[g] = r$\;
      $hit[g] = gene\_symbol(q)$\;
    }
  }
}
\Return{$hit$}\;
\caption{A method for identifying the highest correlated gene to a gene list
within a genomic region}
\end{algorithm}

\begin{algorithm}\label{algorithm:annotation}
\SetLine
\KwIn{All HG-U133\_Plus\_2 probesets $P$}
\KwOut{}
\ForEach{Probeset $p \in P$}{
  $R = sort(C{\prime\prime}_p \ni p)$\;
  $B = Bayesian\_change\_point(R\prime)$\;
  $block = 0$\;
  $offset = 1$\;
  \ForEach{$i \in 1 {\dots} length(B)$}{
    \eIf{$B_i < 0.5$}{
      $block++$\;
    }{
      $block = 0$\;
    }
    \If{$block >= 10$}{
      $offset = i$\;
      Break\;
    }
  }
}
\Return{$hyperGTest(R_1, \dots, R_{offset})$}
\caption{A method for selecting probeset neighbors from correlation data}
\end{algorithm}





\end{document}
